natlas-agent/natlas/threadscan.py

```diff
@@ -120,26 +120,25 @@ def scan(target_data, config):
         for item in screens:
             result.add_screenshot(item)
 
-    if agentConfig["vncScreenshots"] and shutil.which("vncsnapshot") is not None:
-        if "5900/tcp" in result.result["nmap_data"]:
-            if (
-                screenshots.get_vnc_screenshots(
-                    target, scan_id, agentConfig["vncScreenshotTimeout"]
-                )
-                is True
-            ):
-
-                screenshotPath = f"{data_dir}/vncsnapshot.{scan_id}.jpg"
-                if os.path.isfile(screenshotPath):
-                    result.add_screenshot(
-                        {
-                            "host": target,
-                            "port": 5900,
-                            "service": "VNC",
-                            "data": screenshots.base64_image(screenshotPath),
-                        }
-                    )
-                    logger.info(f"VNC screenshot acquired for {result.result['ip']}")
+    if (
+        agentConfig["vncScreenshots"]
+        and "5900/tcp" in result.result["nmap_data"]
+        and screenshots.get_vnc_screenshots(
+            target, scan_id, agentConfig["vncScreenshotTimeout"]
+        )
+    ):
+
+        screenshotPath = f"{data_dir}/vncsnapshot.{scan_id}.jpg"
+        if os.path.isfile(screenshotPath):
```

Comment:
While this code handles errors gracefully, it also seems that we sweep them under the rug. In a future iteration, I'd like to see us try to get the screenshot, then if it fails report that with context (e.g. using Sentry.io).

Reply:
Optional error reporting would be good but I think there's a high chance that this code path isn't even functioning correctly most of the time. x virtual frame buffer running inside a docker container using a tool that frequently doesn't succeed anyways. We'll want to capture_exception so that the scan still gets stored, since screenshots should be considered non-critical functions. Adding exception capturing for both vncsnapshot and web screenshots is worth another ticket, though.